ledger/ledger-consensual/src/main/kotlin/net/corda/v5/ledger/consensual/transaction/ConsensualTransactionVerificationException.kt

```diff
@@ -0,0 +1,16 @@
+package net.corda.v5.ledger.consensual.transaction
+
+import net.corda.v5.base.exceptions.CordaRuntimeException
+import net.corda.v5.crypto.SecureHash
+
+/**
+ * Indicates that some aspect of the transaction named by [transactionId] violates some rules.
+ *
+ * @property transactionId the Merkle root hash (identifier) of the transaction that failed verification.
+ */
+open class ConsensualTransactionVerificationException(
```

Comment:
Why do we need a consensual specific exception?

Reply:
The KDoc of `ConsensualState.verify` says:

```kotlin
/**
     * An override of this function needs to be provided to:
     *  - verify the state's well-formedness
     *  - verify compatibility with the other states of the encapsulating transaction
     *  - check required signing keys
     *  - check the transaction's timestamp.
     *
     * TODO(make services injectable (crypto, etc... CORE-5995)
     *
     * @param ledgerTransaction encapsulating transaction
     *
     * @throws [net.corda.v5.ledger.common.transaction.TransactionVerificationException] if the verification failed.
     *
     */
    fun verify(ledgerTransaction: ConsensualLedgerTransaction)
```

Note, that it says `TransactionVerificationException` is thrown if verification fails.

However, the implementation does this.

```kotlin
private fun verifyStates() {
        transaction.states.map { it.verify(transaction) }
    }
```

Meaning that there is no relation between the type of exception that `ConsensualState.verify` throws and the correct behaviour of the code.

I say we remove `ConsensualTransactionVerificationException`/previously `TransactionVerificationException` altogether because visible benefits. Furthermore, it makes it less like the UTXO version which we generally throw `IllegalStateException`s or `IllegalArgumentException`s, the UTXO version also doesn't care in general what is thrown.

Reply:
I have updated it.